tests/integration/ha/test_storage.py

```diff
@@ -98,21 +103,77 @@ async def test_storage_reuse_after_scale_down(
     assert return_code == 0, "Failed to add unit with storage"
 
     await ops_test.model.wait_for_idle(
-        apps=[app], status="active", timeout=1000, wait_for_exact_units=1
+        apps=[app],
+        status="active",
+        timeout=1000,
+        wait_for_exact_units=2,
     )
 
     # check the storage of the new unit
-    new_unit_id = get_application_unit_ids(ops_test, app)[0]
+    new_unit_id = get_application_unit_ids(ops_test, app)[1]
     new_unit_storage_id = storage_id(ops_test, app, new_unit_id)
     assert unit_storage_id == new_unit_storage_id, "Storage IDs mismatch."
 
     # check if data is also imported
     assert writes_result.count == (await c_writes.count())
     assert writes_result.max_stored_id == (await c_writes.max_stored_id())
 
+    # check if the testfile is still there or was overwritten on installation
+    check_testfile_cmd = f"juju ssh {app}/{new_unit_id} -q sudo ls {testfile}"
+    assert testfile == subprocess.getoutput(check_testfile_cmd)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+@pytest.mark.skip(reason="scaling down to zero and scaling back up doesn't work currently")
+async def test_storage_reuse_after_scale_to_zero(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_writes_runner
+):
+    """Check storage is reused and data accessible after scaling down and up."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    if storage_type(ops_test, app) == "rootfs":
+        pytest.skip(
+            "reuse of storage can only be used on deployments with persistent storage not on rootfs deployments"
+        )
+
+    writes_result = await c_writes.stop()
+
+    # scale down to zero units
+    unit_ids = get_application_unit_ids(ops_test, app)
+    storage_ids = {}
+    for unit_id in unit_ids:
+        storage_ids[unit_id] = storage_id(ops_test, app, unit_id)
+        await ops_test.model.applications[app].destroy_unit(f"{app}/{unit_id}")
+
+    await ops_test.model.wait_for_idle(
+        # app status will not be active because after scaling down not all shards are assigned
+        apps=[app],
+        timeout=1000,
+        wait_for_exact_units=0,
+    )
+
+    # scale up again
+    for unit_id in unit_ids:
+        add_unit_cmd = f"add-unit {app} --model={ops_test.model.info.name} --attach-storage={storage_ids[unit_id]}"
+        return_code, _, _ = await ops_test.juju(*add_unit_cmd.split())
+        assert return_code == 0, f"Failed to add unit with storage {storage_ids[unit_id]}"
+
+    await ops_test.model.wait_for_idle(
+        apps=[app],
+        status="active",
+        timeout=1000,
+        wait_for_exact_units=len(unit_ids),
+    )
+
+    # check if data is also imported
+    assert writes_result.count == (await c_writes.count())
+    assert writes_result.max_stored_id == (await c_writes.max_stored_id())
```

Comment:
Can you also add the following test:

```suggestion
    assert writes_result.max_stored_id == (await c_writes.max_stored_id())

    # Restart it, so we can validate the cluster is still working
    c_writes = ContinuousWrites(ops_test, app, initial_count=writes_result.count)
    await c_writes.start()
    await assert_continuous_writes_increasing(c_writes)
    # final validation
    await assert_continuous_writes_consistency(ops_test, c_writes, app)
```